### ISSUE TITLE
help request: etcd->http://etcd:2379: failed to parse domain

### ISSUE THREAD

**Issue by syl000** [apache/apisix#8800]
### Description

my error log have a warn,I don't know why ,does anyone know it ？

./apisix_log/error.log:

`2023/02/07 02:22:25 [warn] 52#52: *12717338 [lua] v3.lua:829: request_chunk(): http://etcd:2379: failed to parse domain: failed to parse domain. Retrying, context: ngx.timer`

`[error] 76#76: *12281706 [lua] config_etcd.lua:631: no healthy etcd endpoint available, next retry after 2s, context: ngx.timer`

`[warn] 109#109: *10905451 [lua] v3.lua:245: _request_uri(): http://etcd:2379: failed to parse domain: failed to parse domain. Retrying, client: 172.19.0.7, server: , request: "GET /apisix/prometheus/metrics HTTP/1.1", host: "apisix:9091"`

docker-compose.yml
```
etcd:
    image: bitnami/etcd:3.4.15
    restart: always
    volumes:
      - /data/etcd:/bitnami/etcd
    environment:
      ETCD_ENABLE_V2: "true"
      ALLOW_NONE_AUTHENTICATION: "yes"
      ETCD_ADVERTISE_CLIENT_URLS: "http://etcd:2379"
      ETCD_LISTEN_CLIENT_URLS: "http://0.0.0.0:2379"
    ports:
      - "2379:2379/tcp"
    networks:
      apisix:
```

### Environment

- APISIX version (run `apisix version`):3.1.0
- Operating system (run `uname -a`):Debian 4.19.98-1+deb10u1 (2020-04-27) x86_64 GNU/Linux
- OpenResty / Nginx version (run `openresty -V` or `nginx -V`):openresty/1.21.4.1
- etcd version, if relevant (run `curl http://127.0.0.1:9090/v1/server_info`):3.4.15

**Comment by mscb402** [comment]
You can modify `ETCD_ADVERTISE_CLIENT_URLS: "http://etcd:2379"` to `ETCD_ADVERTISE_CLIENT_URLS: "http://0.0.0.0:2379"`

**Comment by monkeyDluffy6017** [comment]
It's werid, the [docker-compose.yml](https://github.com/apache/apisix-docker/blob/master/example/docker-compose.yml) should work, Do you change the file？

**Comment by syl000** [comment]
> It's werid, the [docker-compose.yml](https://github.com/apache/apisix-docker/blob/master/example/docker-compose.yml) should work, Do you change the file？

no,i didn't

**Comment by NokoAnubis** [comment]
I have this error as well, I am using docker compose I have tried changing the advertised url and I've tried installing from source, I've tried using a newer etcd version but I couldn't figure it out. I loose connection temporarily and then I get connection after a few minutes so my configuration works I just think something is broken with apisix for now.

**Comment by monkeyDluffy6017** [comment]
It's luajit's jit problem, we will fix this in next version

**Comment by NokoAnubis** [comment]
> It's luajit's jit problem, we will fix this in next version
alright. when are ya'll thinking about releasing the next version?

**Comment by monkeyDluffy6017** [comment]
It will be released ASAP, probably this week

**Comment by NokoAnubis** [comment]
Awesome!Sent from my iPhoneOn Feb 21, 2023, at 6:48 PM, levy liu ***@***.***> wrote:﻿
It will be released ASAP, probably this week

**Comment by wofr** [comment]
Any update on this?  (Seems to be the same issue as desribed here [https://github.com/apache/apisix/issues/8652](https://github.com/apache/apisix/issues/8652) )

**Comment by monkeyDluffy6017** [comment]
It's fixed, rebuild your apisix-base: https://github.com/api7/apisix-build-tools

**Comment by okkervil** [comment]
> 

The reason is?

**Comment by JanEgner** [comment]
Though I cannot reproduce the issue anymore, with any version (why ever), I believe the version with the fix is 2.15.3, according to the commit logs. This seems to be the actual fix: https://github.com/api7/apisix-build-tools/commit/b3805a0bef56abb1cd0e950437168612272a4681

**Comment by wklken** [comment]
according to the build-apisix-base.sh in apisix-build-tools

fixed in 2.15.3 / 3.2.0 
- https://github.com/api7/apisix-build-tools/blob/apisix/2.15.3/build-apisix-base.sh#L115
- https://github.com/api7/apisix-build-tools/blob/apisix/3.2.0/build-apisix-base.sh#L114-L120

not fixed in 2.15.1 / 2.15.2 / 3.0.0 / 3.1.0

**Comment by monkeyDluffy6017** [comment]
This issue has been fixed in the latest luajit.
We only backport in the latest version (2.15.3 / 3.2.0) under 2.15.x and 3.x.x. 
For other subversions, please refer to the fix: https://github.com/api7/apisix-build-tools/commit/b3805a0bef56abb1cd0e950437168612272a4681

**Comment by zyg166** [comment]
> This issue has been fixed with the latest luajit, it's some kind of jit bug. We only backported in the latest version (2.15.3 / 3.2.0) under 2.15.x and 3.x.x. For other subversions, please refer to the fix: [api7/apisix-build-tools@b3805a0](https://github.com/api7/apisix-build-tools/commit/b3805a0bef56abb1cd0e950437168612272a4681)

I still got errors in my container(I have tried build with lasted version and rebuild apisix-base)

> apisix  | 2023/05/26 05:41:00 [error] 8#8: *72 [lua] resolver.lua:80: parse_domain(): failed to parse domain: etcd, error: failed to query the DNS server: dns server error: 3 name error, client: unix:, server: , request: "POST /v3/kv/range HTTP/1.1", host: "127.0.0.1"
apisix  | 2023/05/26 05:41:00 [error] 8#8: *72 [lua] conf_server.lua:131: resolve_servers(): dns resolver resolves domain: etcd error: failed to query the DNS server: dns server error: 3 name error, client: unix:, server: , request: "POST /v3/kv/range HTTP/1.1", host: "127.0.0.1"
apisix  | 2023/05/26 05:41:00 [error] 8#8: *72 [lua] conf_server.lua:99: balancer(): failure in conf server: no host allowed while connecting to upstream, client: unix:, server: , request: "POST /v3/kv/range HTTP/1.1", host: "127.0.0.1"

- APISIX Version: 3.3.0

- Operating system:  Linux apisix 3.10.0-1127.el7.x86_64  SMP Tue Mar 31 23:36:51 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

- OpenResty / Nginx version: openresty/1.21.4.1
